starboard/shared/starboard/player/filter/audio_frame_discarder.cc

```diff
@@ -41,6 +35,8 @@ void AudioFrameDiscarder::OnInputBuffers(const InputBuffers& input_buffers) {
     });
   }
 
+  // Add an extra check here to ensure that |input_buffer_infos_| won't grow
```

Comment:
Unlike the previous check, this one won't protect from OOM in production as it's a SB_DCHECK().

Reply:
The original code was added to sure that the code won't lead to a crash when |input_buffer_infos_| keeps growing without elements being removed.  That said, any such crashes should be caught during canary stage.